openshift_day2_configurator/configurators/nodes.py

```diff
@@ -0,0 +1,611 @@
+                                }
+                            ]
+                        },
+                    }
+                },
+            ),
+            logger=logger,
+        )
+    }
+
+    if not (
+        image_registries_updated_dict := wait_for_machine_config_pool_to_update(
+            client=client,
+            nodes_type=nodes_type,
+            machine_config_name=search_registries_machine_config_name,
+            node_operation_message=image_registries_desc_message,
+            error_message=f"Failed to configure images registry setting on {nodes_type} nodes",
+        )
+    )[image_registries_desc_message]["res"]:
+        return image_registries_updated_dict
+
+    logger.info(f"Configured image registry setting on {nodes_type} nodes successfully.")
+
+    return image_registries_task_dict
+
+
+def adding_extensions_to_rhcos_on_nodes(
+    client: DynamicClient,
+    logger: logging.Logger,
+    nodes_type: str,
+) -> Dict[str, Dict[str, Union[str, bool]]]:
+    rhcos_extensions_dec_message: str = adding_extensions_to_rhcos_on_nodes_message(nodes_type=nodes_type)
+    logger.debug(rhcos_extensions_dec_message)
+
+    rhcos_extension_name: str = "usbguard"
+
+    machine_config_name = f"80-{nodes_type}-extensions"
+    rhcos_extensions_task_dict = {
+        rhcos_extensions_dec_message: create_ocp_resource(
+            ocp_resource=MachineConfig(
+                client=client,
+                name=machine_config_name,
+                label={MACHINE_CONFIGURATION_ROLE_LABEL: nodes_type},
+                config={"ignition": {"version": IGNITION_VERSION}},
+                extensions=[rhcos_extension_name],
+            ),
+            logger=logger,
+        )
+    }
+
+    if not (
+        rhcos_extensions_updated_dict := wait_for_machine_config_pool_to_update(
+            client=client,
+            nodes_type=nodes_type,
+            machine_config_name=machine_config_name,
+            node_operation_message=rhcos_extensions_dec_message,
+            error_message=f"Failed to add {rhcos_extension_name} extension to rhcos on {nodes_type} nodes",
+        )
+    )[rhcos_extensions_dec_message]["res"]:
+        return rhcos_extensions_updated_dict
+
+    logger.info(f"Added {rhcos_extension_name} extension to rhcos on {nodes_type} nodes successfully.")
+
+    return rhcos_extensions_task_dict
+
+
+def loading_custom_firmware_blobs_on_nodes(
+    client: DynamicClient,
+    logger: logging.Logger,
+    nodes_type: str,
+    firmware_files_dir: str,
+    firmware_blob_file: str,
+) -> Dict[str, Dict[str, Union[str, bool]]]:
+    def assert_butane_params_valid() -> None:
+        assert os.path.exists(
+            f"{firmware_files_dir}/{firmware_blob_file}"
+        ), f"Given firmware blob file {firmware_blob_file} does not exist under {firmware_files_dir} directory."
+
+        assert shutil.which("butane") is not None, "Butane CLI is not installed."
```

Comment:
should not raise; should return a dict with the failure to be printed in the table

Reply:
done